dockstore-webservice/src/main/java/io/dockstore/webservice/core/DescriptionMetrics.java

```diff
@@ -0,0 +1,64 @@
+package io.dockstore.webservice.core;
+
+public class DescriptionMetrics {
+
+    private final long descriptionLength;
+
+    private final long entropy;
+
+    private final long wordCount;
+
+    /**
+     * Calculates various description metrics based on the input description's content.
+     * @param descriptionContent String content of a description
+     */
+    public DescriptionMetrics(String descriptionContent) {
+        if (descriptionContent == null) {
+            this.descriptionLength = 0;
+            this.entropy = 0;
+            this.wordCount = 0;
+        } else {
+            this.descriptionLength = descriptionContent.length();
+            this.entropy = this.calculateEntropy(descriptionContent);
+            this.wordCount = this.calculateWordCount(descriptionContent);
+        }
+    }
+
+    /**
+     * Calculates the entropy of the input description. Entropy, in this case, is represented as the
+     * number of distinct characters within the description.
+     * @param descriptionContent String content of a description
+     */
+    public long calculateEntropy(String descriptionContent) {
+        return descriptionContent.chars().distinct().count();
```

Comment:
Seems a little too simple
How about https://codereview.stackexchange.com/a/909 which had the same observation (might be a library for this)

Reply:
(My information theory is a little rough) I'm concerned we wont be able to effectively leverage Shannon entropy. I'm not sure how to formalize a grading/evaluation metric, as the ideal entropy for a string is in the range `[0, log_2(n)]`, where n is the number of characters in the dictionary.
1. 0 entropy is a sequence of only 1 character (i.e. "aaaaaa" or "b")
2. log_2(n) is a sequence of n distinct characters that all occur with the same frequency (i.e. "abcd" or "aabbccdd" both have a Shannon entropy of 2).

(Depending on how you calculate, the 'true' entropy of english, defined over 27 characters, it's about 2 - 5 bits/char of information, which we could use as a rough estimate, although we have a wider character set)

For now, I can add the Shannon entropy and the max possible entropy metrics, but just wanted to raise potential concerns for how we'd actually use them.

Reply:
Fair, I'm not sure either. 
Was more observing that counting distinct characters seems to max out at 26 characters plus assorted whitespace and punctuation, at least for descriptions in English. Noticed that answer was by Jeff Atwood who works on Discourse https://www.discourse.org/team.html 

That said, noticed that discourse just uses character count anyway, so ignore this and we could probably just copy them. 
![Screenshot from 2021-07-07 11-00-24](https://user-images.githubusercontent.com/1730679/124782921-a2e11f00-df12-11eb-900c-36ce1e9625f8.png)